packages/compiler-cli/src/ngtsc/transform/src/compilation.ts

```diff
@@ -495,6 +500,7 @@ export class TraitCompiler implements ProgramTypeCheckAdapter {
 
   extendedTemplateCheck(sf: ts.SourceFile, extendedTemplateChecker: ExtendedTemplateChecker):
       ts.Diagnostic[] {
+    if (this.compilationMode === CompilationMode.LOCAL) return [];
```

Comment:
Compiler style is to avoid single-line conditions like this (and always use braces):
```suggestion
    if (this.compilationMode === CompilationMode.LOCAL) {
      return [];
    }
```